test/MaticTokenVesting.test.js

```diff
@@ -0,0 +1,147 @@
+const Vesting = artifacts.require("./contracts/MaticTokenVesting.sol")
+const BigNumber = web3.utils.BigNumber
+
+const toWei = web3.utils.toWei
+async function increaseBlockTime(seconds) {
+  return web3.currentProvider.send(
+    {
+      jsonrpc: "2.0",
+      method: "evm_increaseTime",
+      params: [seconds],
+      id: new Date().getTime()
+    },
+    () => {}
+  )
+}
+async function mineOneBlock() {
+  return web3.currentProvider.send(
+    {
+      jsonrpc: "2.0",
+      method: "evm_mine",
+      id: new Date().getTime()
+    },
+    () => {}
+  )
+}
+
+async function assertRevert(promise) {
+  try {
+    const tx = await promise
+    const receipt = await web3.eth.getTransactionReceipt(tx.tx)
+    if (receipt.gasUsed >= 6700000) {
+      return
+    }
+  } catch (error) {
+    const invalidOpcode = error.message.search("revert") >= 0
+    assert(invalidOpcode, "Expected revert, got '" + error + "' instead")
+    return
+  }
+  assert.ok(false, 'Error containing "revert" must be returned')
+}
+
+require("chai")
+  .use(require("chai-bignumber")(BigNumber))
+  .should()
+
+contract("Token", async accounts => {
+  let owner = accounts[0]
+  let token
+  let vesting
+  let totalSupply
+
+  before(async function() {
+    totalSupply = web3.utils.toBN(10000000000)
+    totalSupply = totalSupply.mul(web3.utils.toBN(10 ** 18))
+    token = await Token.new("Matic test", "MATIC", 18, totalSupply, {
+      from: owner
+    })
+    const total = await token.totalSupply.call()
+    vesting = await Vesting.new(token.address, { from: owner })
+    await token.transfer(vesting.address, total, { from: owner })
+  })
+
+  describe("Token Vesting", function() {
+    it("should test vesting for day zero release time", async function() {
+      const vestingContractBalance = await token.balanceOf.call(vesting.address)
+      assert.equal(vestingContractBalance.toString(), totalSupply.toString())
+      let balance = await token.balanceOf.call(accounts[0])
+      assert.equal(balance.toString(), "0")
+      let amount = toWei("1900000000")
+      await vesting.release(1)
+      balance = await token.balanceOf.call(accounts[0])
+      assert.equal(balance.toString(), amount.toString())
+      amount = toWei((117990560 + 1900000000).toString())
+      await vesting.release(2)
+      balance = await token.balanceOf.call(accounts[0])
+      assert.equal(balance.toString(), amount.toString())
+    })
+
+    it("should test token vesting for userX", async function() {
+      const amount = toWei("10")
+      let block = await web3.eth.getBlock("latest")
+      let blockTime = block.timestamp
+      let time = new Date(blockTime)
+      time.setMinutes(time.getMinutes() + 6)
+      time = +time
+      await vesting.addVesting(accounts[1], time.toString(), amount, {
+        from: accounts[0]
+      })
+      await token.transfer(vesting.address, amount)
+      const balance = await vesting.vestingAmount(28)
```

Comment:
can you use the vesting ID derived from addVesting in line 87 above? That way it's more obvious what you did, also we don't want the test to break as soon as we add/remove a vesting in the contract constructor.